SU-SWS/decanter
# Different way of setting up modular spacing

# READY FOR REVIEW

# Summary
- Use the nested sass map approach for modular spacing allowing for easier finetuning and updating of individual values.
- New sass map `$su-responsive-spacing-map` in `variables/core/_responsive-spacing.scss`
- Add new mixin `@responsive-spacing` instead of replacing the old one per Decanter meeting discussion.
- Consider removing the negative modular spacing steps, since the values are 1px apart between adjacent device sizes. Maybe no need for modular spacing at those small ends (just pick the middle value for all devices)?
- This PR also includes some helper functions, e.g., a "map-deep-get" function that CSS Tricks has featured to retrieve values from nested sass maps.

# Needed By (Date)
- When does this need to be merged by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Pull this branch
1. Apply `@include responsive-spacing('padding', 4 3 null 10);` for example to a component
1. Apply `@include responsive-spacing('margin', 8);` for example to a component
1. Verify that it's pulling from the new responsive spacing value in the below sass map (it should render 3 set of spacing values for the 3 breakpoint ranges - xs-sm, md-xl (768px+), 2xl (1500px+))
![Screen Shot 2020-10-08 at 5 14 04 PM](https://user-images.githubusercontent.com/42749717/95527260-b4989480-0989-11eb-8ffa-f0309dbfb18e.png)


# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- #724

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
